src/sys/redox/selector.rs

```diff
@@ -0,0 +1,206 @@
+use event::Event;
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    fs::File,
+    io::prelude::*,
+    mem,
+    os::unix::io::{AsRawFd, RawFd},
+    slice,
+    sync::{
+        Mutex,
+        atomic::{AtomicUsize, Ordering, ATOMIC_USIZE_INIT}
+    },
+    time::Duration
+};
+use syscall::{self, CLOCK_MONOTONIC, EVENT_READ, EVENT_WRITE};
+use {io, Ready, PollOpt, Token};
+
+const TIMEOUT_TOKEN: Token = Token(::std::usize::MAX - 1);
```

Comment:
Documentated somewhere that this is not a valid token for use by the user of this crate.

Reply:
Is it ever documented that you can't use the awakener, `usize::MAX`?